### ISSUE TITLE
Compiler error with Xcode 10.2 (10E125) with Apollo cocoapod 0.9.5

### ISSUE THREAD

**Issue by commanda** [apollographql/apollo-ios#485]
Compiler error:
```
/Users/amanda/Playpen/Fresh/Pods/Apollo/Sources/Apollo/Collections.swift:65:42: Cannot assign value of type 'EnumeratedSequence<[Key]>.Iterator' to type 'EnumeratedSequence<IndexingIterator<Array<Key>>>.Iterator'
```

Repro steps:
* Update to Xcode 10.2 (10E125)
* Create a fresh Xcode project
* `pod init`
* Add `pod Apollo` to the Podfile
* `pod install`
* See that cocoapods installed `Apollo (0.9.5)`
* Open the .xcworkspace
* Attempt to compile the project 
* See the compiler error

I tried setting the Swift language version to 4, 4.2, or 5, and also tried setting the same on the Apollo project, but I still got the compiler error.

Podfile:
```
# Uncomment the next line to define a global platform for your project
platform :ios, '9.0'

target 'Fresh' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for Fresh
  pod 'Apollo'

end
```

**Comment by commanda** [comment]
When I point my podfile at the latest from master at github instead of using the most recent release, the compiler error is fixed.
`pod 'Apollo', :git => 'https://github.com/apollographql/apollo-ios.git'`
I think this means a new release is due?

**Comment by philipengberg** [comment]
Pointing to the `master` branch certainly works, but for me at least, it results in insanely slow compile times (from normally 15 seconds 👉 minutes). Now, I of course cannot say for sure that this is caused by Apollo specifically, but this is literally the only change I've made to get my project to compile in Xcode 10.2.

**Comment by jnutting** [comment]
Yes, and pointing at master is never fun in the long run. It would really be great to have an official release that works with Swift 5!

**Comment by martijnwalraven** [comment]
We're waiting for Travis to update to Xcode 10.2, so we can land #475.

**Comment by philipengberg** [comment]
... The intricate dependency graph of the World Wide Web 😅

**Comment by GiantBlue76** [comment]
What's your ETA on this?  I need this to work because I have a dependency that is failing.  Thanks!

**Comment by jnutting** [comment]
If you're in a really tight spot, @GiantBlue76, you can either point at master or do a temporary local fix - just change the type of the variable that it's complaining about in the 0.9.5 code to match what the compiler expects. I did so, and everything worked just fine.

**Comment by MaxDesiatov** [comment]
FIY Travis does support Xcode 10.2, I'm using it in graphql-community/ApolloAlamofire#3 test builds. I've just added created draft #491 in Apollo repository, but not sure how to start Travis build for that PR, can anyone help please?

**Comment by martijnwalraven** [comment]
@MaxDesiatov I don't think that is Xcode 10.2 final yet, see https://travis-ci.community/t/update-to-xcode-10-2-release/2809/2. As far as I know, that's why #475 is still failing.

**Comment by vandadnp** [comment]
Time to start using CircleCI perhaps? Travis has historically been quite slow in their support of new macOS/Xcode versions. CircleCI as of right now already supports Xcode 10.2!

**Comment by commanda** [comment]
@vandadnp Nope, CircleCI is still on beta 3 of Xcode 10.2. I'm waiting on them to get the GM too.

**Comment by vandadnp** [comment]
@commanda you're right. They didn't explicitly state that it's a beta but they did provide the build number of `10P99q` which *is* in fact beta!

**Comment by vandadnp** [comment]
I contacted Travis' support and this is their reply:

```Thank you for getting in touch!

We're hopefully aiming to have the GA version of Xcode 10.2 available by the end of the week. I'd recommend keeping an eye on https://blog.travis-ci.com/ as we'll most likely make an announcement when it's available.

Hopefully, this helps. Don't hesitate to get in touch if you have any questions or concerns. We'd be happy to help.

Sincerely,
Carla```

🤞

**Comment by GiantBlue76** [comment]
> If you're in a really tight spot, @GiantBlue76, you can either point at master or do a temporary local fix - just change the type of the variable that it's complaining about in the 0.9.5 code to match what the compiler expects. I did so, and everything worked just fine.

Yep, but it doesn't help me for transient dependencies.  In other words, I have a pod that uses Apollo, but then I have an application that depends on that pod.  So that application just uses the podspec to manage the dependency, therefore it pulls the latest version.  Everytime I do a pod update, I need to change the code myself.  It's also a bit of a pain when attempting to document how to set up the project for the rest of our co-workers :D.  Glad that it will be officially released soon.  Thanks!

**Comment by bradleycwells** [comment]
Is there any way I can be notified on an update for this release? I got Apollo as an dependent in my podspec and it fails when I do  **pod repo push**

**Comment by gbrixey** [comment]
@bradleycwells If you have an RSS feed you can subscribe to https://github.com/apollographql/apollo-ios/releases.atom

**Comment by ileitch** [comment]
GitHub support watching only for releases:
<img width="313" alt="Screen Shot 2019-03-29 at 2 34 22 PM" src="https://user-images.githubusercontent.com/48235/55236233-cb581a00-522f-11e9-8a26-95082abfaf10.png">

**Comment by karlisl** [comment]
@martijnwalraven Update is out! https://blog.travis-ci.com/2019-03-29-xcode-10-2-gm-is-now-available

**Comment by victormihaita** [comment]
Since the update for travis is out, do we have any news for the new pod version?

**Comment by martijnwalraven** [comment]
@victormihaita I merged #475, but I wanted to get #466 in before publishing a new release.

**Comment by GiantBlue76** [comment]
This is great news.  Is the release going to be out today?

**Comment by martijnwalraven** [comment]
This should be fixed in 0.10.0.

**Comment by niteshtak** [comment]
when the release will go out for XCode 10.2 support ?

**Comment by martijnwalraven** [comment]
@niteshtak I just released it!

**Comment by niteshtak** [comment]
@martijnwalraven I noticed the build time increased significantly after the `0.10.0` release :(

**Comment by philipengberg** [comment]
Exactly as I noted with the pre-release https://github.com/apollographql/apollo-ios/issues/485#issuecomment-477130636

**Comment by martijnwalraven** [comment]
@niteshtak @philipengberg I have no idea what could be causing this, so all insight is welcome!

**Comment by philipengberg** [comment]
Me neither, and I don't know if Xcode 10.2 itself has equal fault in it. I've reverted to Xcode 10.1, since I won't have time to troubleshoot in a foreseeable future 😞

**Comment by antonbremer** [comment]
> Me neither, and I don't know if Xcode 10.2 itself has equal fault in it. I've reverted to Xcode 10.1, since I won't have time to troubleshoot in a foreseeable future 😞

Thanks for all the hard work you put in it @martijnwalraven . Likewise @philipengberg unfortunately I am not in a position to help troubleshoot at the moment. I will also roll back to Xcode 10.1. Keep up the good work though!

**Comment by Ded77** [comment]
I haved the build time issue too with Xcode 10.2 but It's fixed for me with Xcode 10.2.1 😊
